docs/jobs-queue/jobs.mdx

```diff
@@ -47,6 +47,186 @@ const createdJob = await payload.jobs.queue({
+            input: {
+              imageId: value,
+            },
+          })
+        }
+      },
+    ],
+  },
+}
+```
+
+#### From Custom Endpoints
+
+Queue jobs from your API routes:
+
+```ts
+export const POST = async (req: PayloadRequest) => {
+  const job = await req.payload.jobs.queue({
+    workflow: 'generateMonthlyReport',
+    input: {
+      month: new Date().getMonth(),
+      year: new Date().getFullYear(),
+    },
+  })
+
+  return Response.json({
+    message: 'Report generation queued',
+    jobId: job.id,
+  })
+}
+```
+
+#### From Server Actions
+
+Queue jobs from Next.js server actions:
+
+```ts
+'use server'
+
+import { getPayload } from 'payload'
+import config from '@payload-config'
+
+export async function scheduleEmail(userId: string) {
+  const payload = await getPayload({ config })
+
+  await payload.jobs.queue({
+    task: 'sendEmail',
+    input: { userId },
+  })
+}
+```
+
+### Job Options
+
+When queuing a job, you can pass additional options:
+
+```ts
+await payload.jobs.queue({
+  task: 'sendEmail',
+  input: { userId: '123' },
+
+  // Schedule the job to run in the future
+  waitUntil: new Date('2024-12-25T00:00:00Z'),
+
+  // Assign to a specific queue
+  queue: 'high-priority',
+
+  // Add custom metadata for tracking
+  log: [
+    {
+      message: 'Email queued by admin',
+      createdAt: new Date().toISOString(),
+    },
+  ],
+})
+```
+
+**Common options:**
+
+- `waitUntil` - Schedule the job to run at a specific date/time in the future
+- `queue` - Assign the job to a specific queue (defaults to `'default'`)
+- `log` - Add custom log entries for debugging or tracking
+- `req` - Pass the request context for access control
+
+#### Check Job Status
+
+After queuing a job, you can check its status:
+
+```ts
+const job = await payload.jobs.queue({
+  task: 'processPayment',
+  input: { orderId: '123' },
+})
+
+// Later, check the job status
+const updatedJob = await payload.findByID({
+  collection: 'payload-jobs',
+  id: job.id,
+})
+
+console.log(updatedJob.completedAt) // When it finished
+console.log(updatedJob.hasError) // If it failed
+console.log(updatedJob.taskStatus) // Details of each task
+```
+
+#### Job Status Fields
+
+Each job document contains:
+
+```ts
+{
+  id: 'job_123',
+  taskSlug: 'sendEmail',        // Or workflowSlug for workflows
+  input: { userId: '123' },     // The input you provided
+  completedAt: '2024-01-15...',  // When job completed (null if pending)
+  hasError: false,              // True if job failed
+  totalTried: 1,                // Number of attempts
+  processing: false,            // True if currently running
+  taskStatus: {                 // Status of each task (for workflows)
+    sendEmail: {
+      '1': {
+        complete: true,
+        output: { emailSent: true }
+      }
+    }
+  },
+  log: [                        // Execution log
+    {
+      message: 'Job started',
+      createdAt: '...'
+    }
+  ]
+}
+```
+
 #### Access Control
```

Comment:
Let's mention under access control that you should use these functions specifically and not rely on the collection overrides since that pattern may go away in the future